streaming/receiver.c

```diff
@@ -51,47 +162,264 @@ static void rrdpush_receiver_thread_cleanup(void *ptr) {
 
 #include "../collectors/plugins.d/pluginsd_parser.h"
 
-PARSER_RC streaming_timestamp(char **words, void *user, PLUGINSD_ACTION *plugins_action)
+static void enqueue_replication_req(struct receiver_state *rpt, RRDHOST *host, char *st_id, time_t start, time_t end)
 {
+    struct replication_req req;
+
+    req.host = host;
+    req.st_id = st_id;
+    req.start = start;
+    req.end = end;
+    receiver_tx_enq_cmd(rpt, &req);
+}
+
+
+void send_replication_req(RRDHOST *host, char *st_id, time_t start, time_t end) {
+    char message[RRD_ID_LENGTH_MAX+60];
+    snprintfz(message, RRD_ID_LENGTH_MAX + 60, "REPLICATE %s %ld %ld\n", st_id, start, end);
```

Comment:
The second argument of this function is not using our pattern, it needs to be updated for `RRD_ID_LENGTH_MAX + 59`, or the variable declaration needs to have one more byte.